### PR TITLE
feat: APP-19 generate project slug based on project name

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-19?focusedCommentId=10108

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. From https://deploy-preview-2335--regen-marketplace.netlify.app/ login and create a new project
2. Go through the creation process. After publishing you should see that the project slug has been generated based on the project name
3. Create another project with the same name, the generated slug should be the same but suffixed with "-2" to avoid any duplicate slugs
4. Edit a published project name, it should not update the project slug

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/components/organisms/BasicInfoForm/BasicInfoForm.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import React, { useEffect } from 'react';
+import React, { useCallback, useEffect } from 'react';
 import { useFormState } from 'react-hook-form';
+import { useParams } from 'react-router-dom';
 import { Box } from '@mui/material';
 import { ERRORS, errorsMapping } from 'config/errors';
 import { useSetAtom } from 'jotai';
+import slugify from 'slug';
```
Comment (blushi) [inline web-marketplace/src/components/organisms/BasicInfoForm/BasicInfoForm.tsx:7]: I chose to use this existing `slug` library: https://www.npmjs.com/package/slug

**Comment by erikalogie** [comment]
If I create a project and then try to create another one with the same name, I get the red "Sorry, something went wrong" banner after clicking "save & next" on the basic info screen.

**Comment by erikalogie** [comment]
Also when I try to update the name of an existing project "Just another project name", it says it saves but then just reverts to the previous name after save

**Comment by blushi** [comment]
> If I create a project and then try to create another one with the same name, I get the red "Sorry, something went wrong" banner after clicking "save & next" on the basic info screen.


> Also when I try to update the name of an existing project "Just another project name", it says it saves but then just reverts to the previous name after save


this should be fixed now

**Review by aaronc — APPROVED** [review]
utACK

**Comment by blushi** [comment]
Sorry @erikalogie I've merged this before you could test this again, could you test again and let me know if that's ok? If you see anything, I can address in a follow-up PR

**Comment by erikalogie** [comment]
Yes this is working properly now, thanks

**Pull request merged**